dvc/output/local.py

```diff
@@ -23,6 +26,17 @@ def __init__(self, stage, path, *args, **kwargs):
 
         super().__init__(stage, path, *args, **kwargs)
 
+    @cached_property
+    def remote(self):
+        if self._remote:
+            return self._remote
+
+        work_tree = None
+        if self.repo and self.is_in_repo and is_working_tree(self.repo.tree):
+            work_tree = self.repo.tree
+        tree = LocalRemoteTree(self.repo, {}, work_tree)
+        return LocalRemote(tree)
```

Comment:
Could we do this remote setting in `__init__`?  Just feels a bit weird introducing cached property & _remote mix in both here and base class for this hack.

Reply:
We could, but in that case, we will set remote inside parent constructor and override it here. Thats what I wanted to avoid creating `cached_property`

Reply:
@pared but we can set it after `super()`, no?